playbooks/common/openshift-cluster/upgrades/v3_10/upgrade_control_plane.yml

```diff
@@ -45,6 +46,59 @@
   - set_fact:
       pre_upgrade_complete: True
 
+- import_playbook: label_nodes.yml
+
+# To upgrade, we need masters to be capable of signing certificates
+- hosts: oo_masters
+  serial: 1
+  tasks:
+  - name: Enable core bootstrapping components
+    include_tasks: ../../../../openshift-master/private/tasks/enable_bootstrap.yml
+  - name: Place shim commands on the masters before we begin the upgrade
+    import_role:
+      name: openshift_control_plane
+      tasks_from: static_shim
+
+# TODO: need to verify settings about the bootstrap configs
+# 1. Does network policy match the master config
+
+- hosts: oo_all_hosts
+  tasks:
+  - name: Disable dynamic node configuration until upgrade completes
+    import_role:
+      name: openshift_node_group
+      tasks_from: bootstrap_config
+    vars:
+      r_node_dynamic_config_name: ""
+
+- name: Configure components that must be available prior to upgrade
+  hosts: oo_first_master
+  pre_tasks:
+  - name: Ensure the master bootstrap config has bootstrapping config
+    import_role:
+      name: openshift_node_group
+      tasks_from: upgrade
+  - name: Enable node configuration reconciliation
+    import_role:
+      name: openshift_node_group
+      tasks_from: sync
+  roles:
+  - role: openshift_sdn
+    when: openshift_use_openshift_sdn | default(True) | bool
+
+- name: Update master nodes
```

Comment:
Are we sure we want to upgrade nodes on the masters before upgrade the control plane?

Reply:
For 3.10 yes, we’ll just require that 3.10 nodes work with a 3.9 master.  In future releases this is less of an issue.  What specifically are you concerned with?

Reply:
I just wanted to make sure this code landed in the right spot.  It's the reverse of what we were doing previously.